service/middleware.go

```diff
@@ -118,6 +118,35 @@ func requireUser(onSuccess, onFail http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+type GqlRequest struct {
+	query string
+}
+
+func (uis *UIServer) noMutationsInProdGQLPlaygroundUnlessSuperUser(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		reqBody := GqlRequest{}
+		err := utility.ReadJSON(r.Body, &reqBody)
+		if err != nil {
+			uis.LoggedError(w, r, http.StatusBadRequest, errors.New("request has no body"))
+		}
+		referrer := r.Referer()
+		isReferrerPlaygroundInProd := utility.StringSliceContains([]string{referrer}, "evergreen.mongodb.com/graphql")
+		isMutation := utility.StringSliceContains([]string{reqBody.query}, "mutation")
```

Comment:
I think you can do strings.Contains() which might help otherwise this is just "reqBody.query == mutation"